src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: typeof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach(item => {
+    const grKey = String(item[key]);
```

Comment:
The expression `String(item[key])` assumes that `key` is a valid property of `item`. Ensure that `key` is of type `keyof T` to avoid runtime errors.